www_src/pages/project/touchhandler.js

```diff
@@ -1,69 +1,126 @@
 module.exports = function(positionable) {
-  return {
+
+  var resetTransform = function() {
+    return {
+      x1: false,
+      y1: false,
+      x2: false,
+      y2: false,
+      distance: 0,
+      angle: 0
+    };
+  };
+
+  var transform = resetTransform();
+  var mark = JSON.parse(JSON.stringify(positionable.state));
```

Comment:
fun fact about this kind of deep copying: it might seem like a hack, but is actually the "normal" way to force a deep copy in many languages that predate JS. In Java, for instance, a deep copy in absence of copy constructors is "serialize the instance to bytecode, then read it back in using the same class formats", and done. Instant deep copy.